src/ucp/core/ucp_context.c

```diff
@@ -336,6 +336,13 @@ static ucs_config_field_t ucp_config_table[] = {
    "endpoint.",
    ucs_offsetof(ucp_config_t, ctx.proto_indirect_id), UCS_CONFIG_TYPE_ON_OFF_AUTO},
 
+  {"PEER_EP_MATCH", "y",
+   "Enable peer matching of local endpoints to a remote ones created between\n"
+   "the same pair of workers. If peer matching functionality is disabled, it\n"
+   "makes sure that endpoints are not affected if any remote endpoint is closed,\n"
+   "but can consume more transport resources, and cannot be used with Stream API",
+   ucs_offsetof(ucp_config_t, ctx.peer_ep_match), UCS_CONFIG_TYPE_BOOL},
+
```

Comment:
This does not solve the real issue. We cannot have env that disabled some of the operations.

Reply:
We can handle it better this way. Configuration is known when ucp context is created. So, if this variable is set, we can fail UCP context creation.

Reply:
@shamisp wdyt?

Reply:
You "avoid" API changes but this does create bunch of other issues. Like this is the default mode for all EP, which means memory consumption will go through the roof. Also, context creation error is going to be very obscure.  Do you know what number endpoints the app creates ? 

I don't have immediate alternative solution other that improve scalability of the wire up protocol. 
Overall it appears that all our operations are "one-sided" (ep to worker) and streaming is one of the exceptions. I want to see if we can generalize this in terms of semantics.